bundles/binding/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/MyqBinding.java

```diff
@@ -0,0 +1,393 @@
+
+					if (deviceConfig != null) {
+						GarageDoorDevice garageopener = garageStatus
+								.getDevice(deviceConfig.deviceIndex);
+						if (garageopener != null) {
+
+							State newState = UnDefType.UNDEF;
+							for (Class<? extends State> type : deviceConfig.acceptedDataTypes) {
+								if (StringType.class == type) {
+									newState = new StringType(garageopener
+											.getStatus().getLabel());
+									break;
+								} else if (OpenClosedType.class == type) {
+									if (garageopener.getStatus() == GarageDoorStatus.CLOSED) {
+										newState = OpenClosedType.CLOSED;
+										break;
+									} else {
+										newState = OpenClosedType.OPEN;
+										break;
+									}
+								} else if (UpDownType.class == type) {
+									if (garageopener.getStatus() == GarageDoorStatus.CLOSED) {
+										newState = UpDownType.DOWN;
+										break;
+									} else if (garageopener.getStatus() == GarageDoorStatus.OPEN) {
+										newState = UpDownType.UP;
+										break;
+									}
+								} else if (PercentType.class == type) {
+									if (garageopener.getStatus() != GarageDoorStatus.UNKNOWN) {
+										newState = new PercentType(50);
+										break;
+									}
+								}
+							}
+							if (newState != UnDefType.UNDEF) {
+								eventPublisher
+										.postUpdate(mygItemName, newState);
+							}
+
+							// make sure we are polling frequently
+							if (garageopener.getStatus().inMotion()) {
+								beginRapidPoll(false);
+							}
+						}
+					}
+				}
+			}
+		} catch (InvalidLoginException e) {
+			logger.error("Could not log in, please check your credentials.", e);
+			invalidCredentials = true;
+		} catch (IOException e) {
+			logger.error("Could not connect to MyQ service", e);
+		}
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	public void internalReceiveCommand(String itemName, Command command) {
+		super.internalReceiveCommand(itemName, command);
+		logger.debug("MyQ binding received command '{}' for item '{}'",
+				command, itemName);
+		if (myqOnlineData != null) {
+			computeCommandForItem(command, itemName);
+		} else {
+			logger.warn("Command '{}' for item '{}' not sent", command,
+					itemName);
+		}
+	}
+
+	/**
+	 * Checks whether the command is value and if the deviceID exists then get
+	 * status of Garage Door Opener and send command to change it's state
+	 * opposite of its current state
+	 * 
+	 * @param command
+	 *            The command from the openHAB bus.
+	 * @param itemName
+	 *            The name of the targeted item.
+	 */
+	private void computeCommandForItem(Command command, String itemName) {
+
+		MyqBindingConfig deviceConfig = getConfigForItemName(itemName);
+
+		if (invalidCredentials || deviceConfig == null) {
+			return;
+		}
+
+		try {
+			GarageDoorData garageStatus = myqOnlineData.getGarageData();
+			GarageDoorDevice garageopener = garageStatus
+					.getDevice(deviceConfig.deviceIndex);
+			if (garageopener != null) {
```

Comment:
As mentioned earlier, recommend against implementing toggle logic in binding.  So briefer code:
```
if (command.equals(OnOffType.ON) || command.equals(UpDownType.UP)) {
    myqOnlineData.executeGarageDoorCommand(garageopener.getDeviceId(), 1);
    beginRapidPoll(true);
} else if (command.equals(OnOffType.OFF) || command.equals(UpDownType.DOWN)) {
    myqOnlineData.executeGarageDoorCommand(garageopener.getDeviceId(), 0);
    beginRapidPoll(true);
} else {
    logger.warn("Unknown command {}", command);
}
```